tornado-api/src/main/java/uk/ac/manchester/tornado/api/ImmutableTaskGraph.java

```diff
@@ -164,4 +168,12 @@ void enableProfiler(ProfilerMode profilerMode) {
     void disableProfiler(ProfilerMode profilerMode) {
         taskGraph.disableProfiler(profilerMode);
     }
+
```

Comment:
I think those methods do not belong to the `TaskGraph` (mutable or immutable) but to the `TornadoExecutionPlan`.

Are they required for any internal usage?

Reply:
@jjfumero  did this change

Reply:
The `ExecutionPlan` handles Immutable Task Graphs. And then all Immutable Task Graphs call to action over the task-graph. This is by designed since we can have multiple immutable task-grahps associated to an execution plan. To facilitate the maintainability of this intermediate stage, we keep the same method names as the initial API.

Reply:
ok, makes sense. So they are used for internal purposes. They are not exposed to programmers.